test/space/cu_ntuples_test.py

```diff
@@ -18,70 +18,151 @@
 
 # Imports for common Python 2/3 codebase
 from __future__ import print_function, division, absolute_import
-
 from future import standard_library
 standard_library.install_aliases()
+from builtins import range, str
 
 # External module imports
 import pytest
-import math
+from math import ceil
 import numpy as np
 from numpy import float64
+import scipy as sp
 
 # ODL imports
 import odl
+from odl import ComplexNumbers
 from odl.space.ntuples import FnConstWeighting
 if odl.CUDA_AVAILABLE:
-    from odl.space.cu_ntuples import _CudaFnConstWeighting
+    from odl.space.cu_ntuples import (
+        _CudaFnNoWeighting, CudaFnConstWeighting, CudaFnVectorWeighting,
+        _CudaFnCustomInnerProduct, _CudaFnCustomNorm, _CudaFnCustomDist)
 
-from odl.util.testutils import all_almost_equal, almost_equal, skip_if_no_cuda
+from odl.util.testutils import (all_equal, all_almost_equal, almost_equal,
+                                skip_if_no_cuda)
 
 
 # TODO:
-# * weighted spaces
 # * custom dist/norm/inner
 
 
 def _array(fn):
     # Generate numpy vectors, real or complex or int
     if np.issubdtype(fn.dtype, np.floating):
-        return np.random.randn(fn.size).astype(fn.dtype)
+        return np.random.uniform(-1, 1, fn.size).astype(fn.dtype)
```

Comment:
I changed that in my first attempt to make a [positive array](https://github.com/odlgroup/odl/pull/13/files#diff-7005bf93daa4498c5a50638aa9347a75R72) but then decided on a different implementation. I'll change it back.